### PR TITLE
Nmap

### PR DESCRIPTION
**Pull request by rmikehodges**
Implemented Nmap distributed scanning with evasion
Fixed Socat redirector background process issues

### CODE REVIEW

File: cmd/exec.go [hunk 1]
```diff
@@ -58,22 +70,109 @@ var nmap = &cobra.Command{
 	Use:   "nmap",
 	Short: "execute nmap",
 	Long:  `executes nmap and splits up the job between all of the specified hosts returning the xml files to the specified directory`,
+	Args: func(cmd *cobra.Command, args []string) error {
+		_, err := deployer.ValidatePorts(nmapPorts)
```
Comment (michellehodges) [inline cmd/exec.go]: need to check whether the ports are within a range of legit ports

File: cmd/exec.go [hunk 2]
```diff
@@ -58,22 +70,109 @@ var nmap = &cobra.Command{
...
+
+		list := deployer.ListIPAddresses(marshalledState)
+
+		var instances []deployer.ListStruct
+
+		for _, num := range numsToDeploy {
+			instances = append(instances, list[num])
+		}
+
+		nmapCommands := deployer.SplitNmapCommand(nmapPorts, nmapHostFile, nmapCommand, len(instances), nmapEvasive)
```
Comment (michellehodges) [inline cmd/exec.go]: what does `SpliNmapCommand` do? probably need to consider a better name

File: cmd/exec.go (hunk 1 above)
Comment (rmikehodges) [reply cmd/exec.go]: Fixed

File: cmd/exec.go (hunk 2 above)
Comment (rmikehodges) [reply cmd/exec.go]: Changed to SplitNmapCommandsIntoHosts

**Pull request merged**